docs/flow/modin/core/execution/dask/implementations/pandas_on_dask/index.rst

```diff
@@ -16,3 +42,36 @@ specific for `PandasOnDask` execution.
     partitioning/partition
     partitioning/axis_partition
     partitioning/partition_manager
+
+
+Data Ingress
+''''''''''''
+
+.. image:: /img/pandas_on_dask_data_ingress.svg
+   :align: center
+
+Data Egress
+'''''''''''
+
+.. image:: /img/pandas_on_dask_data_egress.svg
+   :align: center
+
+
+When a user calls any IO function from the ``modin.pandas.io`` module, the `API` layer queries the
+:py:class:`~modin.core.execution.dispatching.factories.dispatcher.FactoryDispatcher` which defines a factory specific for
+the execution, namely, the :py:class:`~modin.core.execution.dispatching.factories.factories.PandasOnDaskFactory`. The factory, in turn,
+exposes the :py:class:`~modin.core.execution.dask.implementations.pandas_on_dask.io.PandasOnDaskIO` class
+whose responsibility is to perform a parallel read/sequential write from/to a file.
+
+When reading data from a CSV file, for example, the :py:class:`~modin.core.execution.dask.implementations.pandas_on_dask.io.io.PandasOnDaskIO` class forwards
+the user query to the :meth:`~modin.core.io.text.CSVDispatcher._read` method of :py:class:`~modin.core.io.text.CSVDispatcher`, where the query's parameters are preprocessed
+to check if they are supported by the execution (defaulting to pandas if they are not) and computes some metadata
+common for all partitions to be read. Then, the file is split into row chunks, and this data is used to launch remote tasks on the Dask workers
+via the :meth:`~modin.core.execution.dask.common.task_wrapper.DaskTask.deploy` method of :py:class:`~modin.core.execution.dask.common.task_wrapper.DaskTask`.
+On each Dask worker, the :py:class:`~modin.core.storage_formats.pandas.parsers.PandasCSVParser` parses data.
+After the remote tasks are finished, additional result postprocessing is performed,
+and a new query compiler with the data read is returned.
+
+Writing data to a file happens sequentially for `pandas on Dask` execution. When writing data to a CSV file, for example, the :py:class:`~modin.core.execution.dask.implementations.pandas_on_dask.io.PandasOnDaskIO` converts a query compiler
```

Comment:
I don't think we should say that writing data to a file is a sequential for pandas on Dask. Actually, that should be performed in parallel. I would prefer to say that writing data is parallel but for now pandas On Dask defaults to pandas for the flow.

Reply:
Currently, this is performed by the similar way as for `Python` engine (sequentially), because `PandasOnDaskIO` has implementations for `read_*` functions only.

Reply:
Yes, I see. That's why I suggest adding a note `for NOW pandas On Dask defaults to pandas for the flow`.

Reply:
Changed

Reply:
I suggest adding some notes on parallel write, i.e. `in general writing data would happen in parallel... but for now it defaults to pandas...` look at pandas on Ray execution. The picture should be correct too.

Reply:
I suggest to leave the current sentences/picture as it is, because we add documentation for current implementation. Developer could be confused when he will read `writing data would happen in parallel`/see the picture with class dependencies for data egress and compare those with the codebase. In my opinion, we can start to mix the `parallel..`/`default to pandas` statements when we have at least one parallel implementation of `write` function for  `pandas on Dask` execution, but we haven't this for now. That's reason why I use only `default to pandas` statement for this doc.

When developer will add the parallel `write` implementation for pandas on Dask, he also should include changes in the docs by our contributing guidelines. 

@vnlitvinov, @devin-petersohn What do you think?

Reply:
Data egress for `pandas on Dask` should be parallel by nature as it is for `pandas on Ray`. I would prefer say about this and add a note that currently data egress for `pandas on Dask` uses default pandas implementation.

Reply:
I agree with @YarShev let's make pandas on Ray and pandas on Dask have the same sentences, with the exception of adding a `..note::` or somehow highlighting the fact that it currently defaults to pandas.

Reply:
Suggestions were applied